lib/src/chart/pie_chart/pie_chart_painter.dart

```diff
@@ -312,14 +312,46 @@ class PieChartPainter extends BaseChartPainter<PieChartData> {
     Path sectionPath,
     CanvasWrapper canvasWrapper,
   ) {
+    // _sectionPaint
+    //   ..setColorOrGradient(
+    //     section.color,
+    //     section.gradient,
+    //     sectionPath.getBounds(),
+    //   )
+    //   ..style = PaintingStyle.fill;
+    // canvasWrapper.drawPath(sectionPath, _sectionPaint);
+
+    // Set up the paint for the section
     _sectionPaint
       ..setColorOrGradient(
         section.color,
         section.gradient,
         sectionPath.getBounds(),
       )
       ..style = PaintingStyle.fill;
-    canvasWrapper.drawPath(sectionPath, _sectionPaint);
+
+    // Apply border radius if specified
+    if (section.borderRadius != BorderRadius.zero) {
+      Rect bounds = sectionPath.getBounds();
+      Path roundedPath = Path()
+        ..addRRect(
+          RRect.fromRectAndCorners(
```

Comment:
This logic doesn't seem right to me.
And when I applied `borderRadius: const BorderRadius.all(Radius.circular(200)),` to our PieChartSample1, I saw this:
<img width="544" alt="image" src="https://github.com/user-attachments/assets/ffddbf13-05fa-43d0-8652-a72abdfa07e0" />

It only works with small values like 8, 16, ...
So it is not stable I would say

Reply:
I think the better idea is just to edit the shape path generation logic here:

https://github.com/imaNNeo/fl_chart/blob/01689ee2feb8ed560c80e16c19bd25f436737cb1/lib/src/chart/pie_chart/pie_chart_painter.dart#L220-L227

We might be able to use [arcTo](https://api.flutter.dev/flutter/dart-ui/Path/arcTo.html), [conicTo](https://api.flutter.dev/flutter/dart-ui/Path/conicTo.html) or [cubicTo](https://api.flutter.dev/flutter/dart-ui/Path/cubicTo.html) in [Path](https://api.flutter.dev/flutter/dart-ui/Path-class.html)